File: frontend/types/todo.ts
export type Todo = {
  id: number;
  title: string;
  detail: string;
  isCompleted: boolean;
  createdAt: string;
  updatedAt: string;
};

export type TodoInput = {
  title: string;
  detail: string;
  isCompleted?: boolean;
};

File: src/main/java/com/example/demo/entity/Todo.java
package com.example.demo.entity;

import jakarta.validation.constraints.NotBlank;

import java.time.OffsetDateTime;

public record Todo(
        Long id,
        @NotBlank String title,
        @NotBlank String detail,
        Boolean isCompleted,
        OffsetDateTime createdAt,
        OffsetDateTime updatedAt
) {
}
